File: programmers/lv2/두큐합같게만들기.py
from collections import deque
def solution(queue1, queue2):
    answer = -1
    queue1 = deque(queue1)
    queue2 = deque(queue2)
    sum1 = sum(queue1)
    sum2 = sum(queue2)
    maxLimit = len(queue1)+len(queue2)+100
    
    cnt = 0
    while(cnt < maxLimit):
        val = 0
        if sum1 > sum2:
            val = queue1.popleft()
            queue2.append(val)
            sum1 -= val
            sum2 += val
        elif sum1 < sum2:
            val = queue2.popleft()
            queue1.append(val)
            sum1 += val
            sum2 -= val
        else:
            answer = cnt
            break
        cnt += 1
    return answer

File: programmers/lv2/두큐합같게만들기.js
function solution(queue1, queue2) {
  var answer = -1;
  const sum = (arr) => arr.reduce((a, b) => a + b);

  let sumA = sum(queue1);
  let sumB = sum(queue2);
  let target = (sumA + sumB) / 2;
  let left = 0,
    right = queue1.length;

  let list = [...queue1, ...queue2];
  let limit = list.length * 3;

  let cnt = 0;
  while (cnt <= limit) {
    if (sumA == target) {
      answer = cnt;
      break;
    } else if (sumA > target) {
      sumA -= list[left];
      left += 1;
    } else {
      sumA += list[right];
      right += 1;
    }
    cnt += 1;
  }

  return answer;
}
